docs/rules.md

```diff
@@ -139,6 +139,7 @@ Rules are sorted by severity.
 | `RESTRICTED_HOMEPAGE_URL`                               | error    | Linking to addons.mozilla.org in `homepage_url` or `developer.url` is not allowed               |
 | `RESTRICTED_PERMISSION`                                 | error    | A permission requires "strict_min_version" to be set to a specific Firefox version              |
 | `EXTENSION_ID_REQUIRED`                                 | error    | The extension ID is mandatory for Manifest Version 3 (and above) extensions.                    |
+| `HIDDEN_NO_BROWSER_ACTION`                              | error    | The `hidden` and `browser_action` (or `action`) properties are mutually exclusive.              |
```

Comment:
@willdurand I'm wondering if we should also do the same for the page_action, in the end we mainly focused our interest on browserAction because it was directly part of details discussed for the other ongoing work, but with those motivations having a page_action wouldn't be that different in the end.

Another one that may not make sense to allow when the addon is hidden is options_ui (at least when there isn't also a open_in_tab set to true).

We may opt to consider these but defer covering some if we think they will need some more discussion, but I thought to mention it in this PR in case we may want to tweak the rule code so that the other ones would also fit in the same rule code.

e.g. we may opt for something like `HIDDEN_NO_ACTION` or `HIDDEN_UI_ACTION` as a rule id if we only intend to use it for browserAction and pageAction

or maybe something like `NO_HIDDEN_WITH_UI`  may be one that we could also use for the options_ui case (if we agree that would also worth).

wdyt?

Reply:
PR updated to restrict `page_action`. I'll file an issue for `options_ui`.